torch/nn/functional.py

```diff
@@ -563,6 +563,37 @@ def batch_norm(input, running_mean, running_var, weight=None, bias=None,
     return f(input, weight, bias)
 
 
+def relation(input, g, f=None, embedding=None, max_pairwise=None):
+    # Batch size, number of objects, feature size
+    b, o, c = input.size()
+    # Construct pairwise indices
+    i = Variable(torch.arange(0, o).long().repeat(o))
+    j = Variable(torch.arange(0, o).long().repeat(o, 1).t().contiguous().view(-1))
```

Comment:
So now we have two arrays, both of size o^2, iterating through all possible pairwise indexes. Why long, btw?